sdk/src/main/java/io/opentelemetry/sdk/trace/Samplers.java

```diff
@@ -259,39 +279,44 @@ public final String getDescription() {
 
   @Immutable
   @AutoValue
-  abstract static class DecisionImpl implements Decision {
+  abstract static class SamplingResultImpl implements SamplingResult {
     /**
-     * Creates sampling decision with probability attribute.
+     * Creates sampling result with probability attribute.
      *
-     * @param decision sampling decision
-     * @param probability the probability that was used for the decision.
+     * @param decision the decision on sampling and recording
+     * @param probability the probability that was used for the samplingResult.
      */
-    static Decision createWithProbability(boolean decision, double probability) {
-      return new AutoValue_Samplers_DecisionImpl(
+    static SamplingResult createWithProbability(Decision decision, double probability) {
+      return new AutoValue_Samplers_SamplingResultImpl(
           decision, Attributes.of(SAMPLING_PROBABILITY.key(), doubleAttributeValue(probability)));
     }
 
     /**
-     * Creates sampling decision without attributes.
+     * Creates sampling result without attributes.
      *
-     * @param decision sampling decision
+     * @param decision sampling samplingResult
      */
-    static Decision createWithoutAttributes(boolean decision) {
-      return new AutoValue_Samplers_DecisionImpl(decision, Attributes.empty());
+    static SamplingResult createWithoutAttributes(Decision decision) {
+      return new AutoValue_Samplers_SamplingResultImpl(decision, Attributes.empty());
     }
 
     /**
-     * Creates sampling decision with the given attributes.
+     * Creates sampling result with the given attributes.
      *
-     * @param decision sampling decision
+     * @param decision sampling decisionq
      * @param attributes attributes. Will not be copied, so do not modify afterwards.
      */
-    static Decision create(boolean decision, Attributes attributes) {
-      return new AutoValue_Samplers_DecisionImpl(decision, attributes);
+    static SamplingResult create(Decision decision, Attributes attributes) {
+      return new AutoValue_Samplers_SamplingResultImpl(decision, attributes);
     }
 
     @Override
-    public abstract boolean isSampled();
+    public abstract Decision getDecision();
+
+    @Override
+    public boolean isSampled() {
```

Comment:
Should we deprecate this? Maybe a followup PR, for the moment just create an issue to not forget about?